blocks/query/edit.tsx

```diff
@@ -169,21 +170,31 @@ export default function Edit({
       );
       path += `&${termQueryArgs}`;
 
-      apiFetch({
-        path,
-      }).then((response) => {
-        setAttributes({ backfillPosts: response as Array<number> });
+      apiFetch({ path }).then((response:any) => {
+        let revisedResponse;
+        if (Array.isArray(response)) {
+          revisedResponse = response.filter((item) => item !== currentPostId);
+        } else if (response.id === currentPostId) {
+          // If the current post is in the response, remove it.
+          revisedResponse = null;
+        } else {
+          revisedResponse = response;
+        }
```

Comment:
the response should always be an array. I'm not sure how to convince typescript of that.